### DISCUSSION TITLE
Query on permissions on .bash_x files

### DISCUSSION THREAD

**Discussion by karlbrown-va** [twpayne/chezmoi#2551] [Q&A]
Hi, two part question:
1) For some reason I think I originally set up my `.bashrc` as private - e.g. `private_dot_bashrc`, even though other files like `.gitconfig` are not private (and I've forgotten why...). Is there a 'best practice' on this (I'm using Mac OS, as the only user on this computer)? What should the permissions on these files be, or does it not matter much?
2) Did something change recently with chezmoi and private files? I did a sync recently and it showed permissions changing on these files from 644 to 600 (e.g. in my `~` directory the files were 644, but chezmoi wanted to change them to 600) Is this a bug that was fixed in the past few months? My other thought is, OSX used to have these as 600, then for some reason they changed them to be 644, and chezmoi noticed this difference.

I'm now running chezmoi v2.26.

**Comment by twpayne** [comment]
> 1. For some reason I think I originally set up my `.bashrc` as private - e.g. `private_dot_bashrc`, even though other files like `.gitconfig` are not private (and I've forgotten why...). Is there a 'best practice' on this (I'm using Mac OS, as the only user on this computer)? What should the permissions on these files be, or does it not matter much?

Different OSes and distributions have different conventions here. If you're on a multiuser system and `.bashrc` contains secrets then you should make sure that `.bashrc` is not readable by others. This can be done by changing the permissions of your home directory to `700` though, you don't need to make individual private.

> 2. Did something change recently with chezmoi and private files?

No, chezmoi has not changed anything related to permissions for a long time. Did you change your umask?
